src/container.ts

```diff
@@ -12,11 +12,14 @@ export type ContainerEventType =
     "layout-saved";
 
 export class WindowEventArgs extends EventArgs {
-    public readonly window: ContainerWindow;
+    public readonly window?: ContainerWindow;
+    public readonly windowId: string;
+    public readonly windowName?: string;
 }
 
 export class LayoutEventArgs extends EventArgs {
-    public readonly layout: PersistedWindowLayout;
```

Comment:
Static events currently don't have a layout instance in the args so changed to optional.